include/bluetooth/mesh/scene_cli.h

```diff
@@ -0,0 +1,293 @@
+	uint32_t remaining_time;
+};
+
+/** Scene register parameters */
+struct bt_mesh_scene_reg {
+	/** Status of the previous operation. */
+	enum bt_mesh_scene_status status;
+	/** Current scene. */
+	uint16_t current;
+	/** Number of entries in the scenes list. */
+	size_t count;
+	/** Optional list of scenes to populate. */
+	uint16_t *scenes;
+};
+
+/** Scene client model instance. */
+struct bt_mesh_scene_cli {
+	/** @brief Status message callback.
+	 *
+	 *  @param[in] cli   Scene client model.
+	 *  @param[in] ctx   Message context parameters.
+	 *  @param[in] state Received scene state.
+	 */
+	void (*status)(struct bt_mesh_scene_cli *cli,
+		       struct bt_mesh_msg_ctx *ctx,
+		       const struct bt_mesh_scene_state *state);
+
+	/** @brief Scene register message callback.
+	 *
+	 *  @param[in] cli Scene client model.
+	 *  @param[in] ctx Message context parameters.
+	 *  @param[in] reg Received scene register.
+	 */
+	void (*reg)(struct bt_mesh_scene_cli *cli, struct bt_mesh_msg_ctx *ctx,
+		    const struct bt_mesh_scene_reg *reg);
+
+	/* Composition data entry pointer. */
+	struct bt_mesh_model *mod;
+	/* Model publication parameters. */
+	struct bt_mesh_model_pub pub;
+	/* Publication message */
+	struct net_buf_simple pub_msg;
+	/* Publication message buffer. */
+	uint8_t buf[BT_MESH_MODEL_BUF_LEN(BT_MESH_SCENE_OP_RECALL,
+					  BT_MESH_SCENE_MSG_MAXLEN_RECALL)];
+	/* Ack context */
+	struct bt_mesh_model_ack_ctx ack;
+	/* Transaction ID */
+	uint8_t tid;
+};
+
+/** @brief Get the current state of a Scene Server.
+ *
+ *  This call is blocking if the @c rsp buffer is non-NULL. The response will
+ *  always be passed to the @ref bt_mesh_scene_cli::status callback.
+ *
+ *  @param[in]  cli Scene client model.
+ *  @param[in]  ctx Message context to send with, or NULL to use the configured
+ *                  publication parameters.
+ *  @param[out] rsp Response buffer, or NULL to keep from blocking.
+ *
+ *  @retval 0 Successfully got the scene state.
+ *  @retval -EALREADY A blocking request is already in progress.
+ *  @retval -EADDRNOTAVAIL A message context was not provided and publishing is
+ *  not configured.
+ *  @retval -EAGAIN The device has not been provisioned.
+ *  @retval -ETIMEDOUT The request timed out without a response.
+ */
+int bt_mesh_scene_cli_get(struct bt_mesh_scene_cli *cli,
+			  struct bt_mesh_msg_ctx *ctx,
+			  struct bt_mesh_scene_state *rsp);
+
+/** @brief Get the full scene register of a Scene Server.
+ *
+ *  This call is blocking if the @c rsp buffer is non-NULL. The response will
+ *  always be passed to the @ref bt_mesh_scene_cli::reg callback.
+ *
+ *  @param[in]  cli Scene client model.
+ *  @param[in]  ctx Message context to send with, or NULL to use the configured
+ *                  publication parameters.
+ *  @param[out] rsp Response buffer, or NULL to keep from blocking. If the
+ *                  @c rsp.scenes parameter points to a valid buffer, it will be
```

Comment:
Probably we need to emphasize that `rsp.count` should set to the size of `rsp.scenes` buffer. Otherwise even if `rsp.scenes` points to a valid buffer, `rsp.count` may not be set and use will get `0` entries :)